src/org/exist/collections/MutableCollection.java

```diff
@@ -1258,6 +1258,9 @@ private void storeXMLInternal(final Txn transaction, final DBBroker broker, fina
         if(info.isCreating()) {
             info.getTriggers().afterCreateDocument(broker, transaction, document);
         } else {
+            final StreamListener listener = broker.getIndexController().getStreamListener(document, StreamListener.ReindexMode.REPLACE_DOCUMENT);
```

Comment:
Nope, they are different. `IndexController` create chain by calling [this](https://github.com/adamretter/exist/blob/c841f6b7d17308be143b1e218a40750e0697264d/src/org/exist/indexing/IndexController.java#L315), but there was no restrictions to avoid `listener` creation by `IndexWorker`.

By current PR you force `IndexWorker` to return same `listener`. That is contract change, no?